examples/user_guide/15-Streaming_Data.ipynb

```diff
@@ -568,29 +568,51 @@
    "outputs": [],
    "source": [
     "%%opts Curve [width=600]\n",
-    "from tornado.ioloop import IOLoop\n",
+    "from tornado.ioloop import PeriodicCallback\n",
     "from tornado import gen\n",
     "\n",
+    "count = 0\n",
     "buffer = Buffer(np.zeros((0, 2)), length=50)\n",
     "\n",
     "@gen.coroutine\n",
     "def f():\n",
-    "    for x in range(500):\n",
-    "        gen.sleep(0.1)\n",
-    "        buffer.send(np.array([[x, np.random.rand()]]))\n",
-    "        \n",
-    "IOLoop.current().add_callback(f)\n",
+    "    global count\n",
```

Comment:
Globals are the best thing to use here?

Reply:
Coroutines don't allow generators so that's not an option, open to other suggestions though.